### PR TITLE
PIM-6342: Display and remove associations gallery view

### PR DESCRIPTION
**Pull request by tamarasaurus**
**Description (for Contributor and Core Developer)**

Changes:
- Add a new datagrid row view for the association grid in the PEF
- Add cross icon instead of unchecking to remove association
- Update associated product datasource to only return is_associated instead of all
- Hide association grid filters
- Add search filter to associated product picker (PIM-7028). 

Bug fixes:
- Fix a bug where the search filter was being rendered twice on the association grid when the item picker is opened (the code that renders the filter was not scoped to an element)
- Fix a similar bug in the pagination rendering

screen shot 2017-12-06 at 11 13 52

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | OK
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | OK
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: features/product/associate_product.feature [hunk 1]
```diff
@@ -47,16 +48,23 @@ Feature: Associate a product
...
     And I check the row "caterpillar_boots"
+    And I save the product
     And I visit the "Cross sell" association type
     And I check the row "similar_boots"
+    And I save the product
     And I visit the "Display products" target
+    And I press the "Add associations" button
     And I check the rows "shoelaces, gray-boots, brown-boots and green-boots"
-    When I save the product
+    And I press the "Confirm" button in the popin
```
Comment (SamirBoulil) [inline features/product/associate_product.feature]: When

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/tab/associated-product-row.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<% if (canRemoveAssociation) { %>
+    <div class="AknGrid-bodyRow-remove AknIconButton AknIconButton--remove AknIconButton--topLeft"></div>
+<% } %>
+<div class="AknGrid-fullImage" style="background-image: url('<%- imagePath %>')"></div>
+<td class="AknGrid-title"><%- label %></td>
```
Comment (SamirBoulil) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/tab/associated-product-row.html]: is the td normal here ? there is no tr nor table ? (maybe they are in other templates. But it seems wierd to have <div> next to <td> tags.

Comment (tamarasaurus) [reply src/Pim/Bundle/EnrichBundle/Resources/public/templates/product/tab/associated-product-row.html]: The parent template is a tr

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-list.js [hunk 3]
```diff
@@ -100,7 +100,7 @@ define(['underscore', 'pim/form', 'oro/mediator', 'oro/tools'],
                     }
                     if (filter.$el.length > 0) {
                         if (filter.isSearch && (this.options.displayedAsColumn === true)) {
-                            $('.search-zone').empty().append(filter.$el.get(0));
+                            $('.search-zone', this.getRoot().$el).empty().append(filter.$el.get(0));
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-list.js]: You should use this.getRoot().$('.search-zone')

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/pagination-input.js [hunk 4]
```diff
@@ -46,6 +46,11 @@ define(
         initialize: function (options) {
             this.appendToGrid = options.appendToGrid;
             this.gridElement = options.gridElement;
+            this.gridName = options.config.gridName;
+
+            if (null === this.gridName || undefined === this.gridName) {
+                throw Error('You must set the gridName for the oro/datagrid/pagination-input');
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datagrid/pagination-input.js]: Maybe we could give a more meaningful error message

**Pull request merged**